### PR TITLE
Add summary to worklow watch

### PR DESCRIPTION
**Pull request by sebneira**
## What was changed
Added workflow execution summary to trace command.
Depends on https://github.com/temporalio/cli/pull/233.

### Example:
Screenshot 2023-05-24 at 18 00 11

## Why?
To provide important information about what workflow execution is traced. This could be extended to include other information as StartTime (or a diff with now).

## Checklist

1. Closes texp upstream
1. How was this tested:
Running temporal (see example provided above)
1. Any docs updates needed?
None

### CODE REVIEW

File: trace/summary.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	res, err := sdkClient.DescribeWorkflowExecution(tcCtx, wfId, runId)
+	if err != nil {
+		return err
+	}
+
+	info := res.GetWorkflowExecutionInfo()
+
+	_, _ = title.Println("Execution summary:")
+	tb := NewTable(os.Stdout)
+	tb.AppendRows(
```
Comment (feedmeapples) [inline trace/summary.go:75]: this seems similar to an existing API, https://github.com/temporalio/cli/blob/f8c84edae78379ad3716ed51c84120c923db9388/cluster/cluster_commands.go#L61

Have you also considered using it?

Comment (feedmeapples) [reply trace/summary.go]: yea it was not documented. Designed this way to accept user's input in a simple form
Let me look into documenting this better

**Review by feedmeapples — APPROVED** [review]

**Comment by feedmeapples** [comment]
pushed few commits to make `PrintWorkflowTrace` easier to read (and address golangci-lint failing on complexity). There is an issue with execution updates after my commits, addressing these and merging

**Comment by sebneira** [comment]
Hi 👋 
Is there any updates on this PR?

Thanks

**Review by bergundy — APPROVED** [review]
I would consider a flag to skip printing the summary but otherwise LGTM.

Thanks.

**Pull request merged**